BASE/src/Microsoft.ApplicationInsights/Metrics/Implementation/AutocollectedMetricsExtraction/DependencyMetricsExtractor.cs

```diff
@@ -128,162 +156,30 @@ public void ExtractMetrics(ITelemetry fromItem, out bool isItemProcessed)
                 return;
             }
 
-            Metric dependencyCallMetric = this.dependencyCallDurationMetric;
-
             //// If there is no Metric, then this extractor has not been properly initialized yet:
-            if (dependencyCallMetric == null)
+            if (this.dependencyCallDurationMetric == null)
             {
                 //// This should be caught and properly logged by the base class:
                 throw new InvalidOperationException(Invariant($"Cannot execute {nameof(this.ExtractMetrics)}.")
                                                   + Invariant($" There is no {nameof(this.dependencyCallDurationMetric)}.")
                                                   + Invariant($" Either this metrics extractor has not been initialized, or it has been disposed."));
             }
 
-            //// Get dependency call success status:
-            bool dependencyFailed = (dependencyCall.Success != null) && (dependencyCall.Success == false);
-            string dependencySuccessString = dependencyFailed ? bool.FalseString : bool.TrueString;
-
-            //// Now we need to determine which data series to use:
-            MetricSeries seriesToTrack = null;
-
-            if (this.MaxDependencyTypesToDiscover == 0)
-            {
-                // (MaxDependencyTypesToDiscover == 0) means we do not group by Dependency Type.
-                // Then, always use "Other" as Dependency Type:
-
-                dependencyCallMetric.TryGetDataSeries(
-                                                    out seriesToTrack,
-                                                    MetricTerms.Autocollection.Metric.DependencyCallDuration.Id,
-                                                    dependencySuccessString,
-                                                    MetricTerms.Autocollection.DependencyCall.TypeNames.Other);
-            }
-            else
+            int i = 0;
+            string[] dimValues = new string[this.dimensionExtractors.Count];
+            foreach (var dim in this.dimensionExtractors)
```

Comment:
dimensionExtractors is a list, I thin you can simplify and join counter `i` and list traversing into a single for loop:
```
for (int i=0; i < dimValues.Length; i++ )
{
   dimValues[i] = this.dimensionExtractors[i].ExtractDimension(dependencyCall);
}
```

It also may end up more performant (doing this for Telemetry Initializers saved me on Iterator as per profile traces).